index.js

```diff
@@ -349,53 +346,47 @@ class Analytics {
       callback(err, data);
     };
 
-    // Don't set the user agent if we're not on a browser. The latest spec allows
-    // the User-Agent header (see https://fetch.spec.whatwg.org/#terminology-headers
-    // and https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/setRequestHeader),
-    // but browsers such as Chrome and Safari have not caught up.
-    const headers = {};
-    if (typeof window === "undefined") {
-      headers["user-agent"] = `analytics-node/${version}`;
-    }
-
     const req = {
-      method: "POST",
-      url: `${this.host}`,
-      auth: {
-        username: this.writeKey,
+      method: 'POST',
+      headers: {
+        // Don't set the user agent if we're not on a browser. The latest spec allows
+        // the User-Agent header (see https://fetch.spec.whatwg.org/#terminology-headers
+        // and https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/setRequestHeader),
+        // but browsers such as Chrome and Safari have not caught up.
+        'User-Agent':
+          typeof window === 'undefined'
+            ? `analytics-node/${version}`
+            : undefined,
+        Authorization:
+          'Basic ' + Buffer.from(`${this.writeKey}:`).toString('base64'),
       },
-      data,
-      headers,
+      body: JSON.stringify(data),
+      retries: 3,
+      retryDelay: (attempt, error, response) => Math.pow(2, attempt) * 1000, // Exponential backoff - 1000, 2000, 4000
```

Comment:
The prior function used a different formula.
https://github.com/softonic/axios-retry/blob/8d6dff054c087031e7e93dc7bf49ed1c8f20cba1/es/index.js#L77-L81

Also note that the prior function started the retryNumber from 1 while this new function receives 0 for the first retry attempt.
https://github.com/softonic/axios-retry/blob/8d6dff054c087031e7e93dc7bf49ed1c8f20cba1/es/index.js#L200-L201

Reply:
updated the function, thanks for pointing this out! I also changed the delay factor from 100 ms => 200 ms so that there's a bit more time for network issues to resolve themselves.